kubevirt_api/datadog_checks/kubevirt_api/check.py

```diff
@@ -0,0 +1,226 @@
+# (C) Datadog, Inc. 2024-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import ipaddress
+from typing import Any  # noqa: F401
+from urllib.parse import urlparse
+
+from datadog_checks.base import OpenMetricsBaseCheckV2
+from datadog_checks.base.checks.openmetrics.v2.transform import get_native_dynamic_transformer
+
+from .kube_client import KubernetesAPIClient
+from .metrics import METRICS_MAP
+
+
+class KubeVirtApiCheck(OpenMetricsBaseCheckV2):
+    __NAMESPACE__ = "kubevirt_api"
+    DEFAULT_METRIC_LIMIT = 0
+
+    def __init__(self, name, init_config, instances):
+        super(KubeVirtApiCheck, self).__init__(name, init_config, instances)
+        self.check_initializations.appendleft(self._parse_config)
+        self.check_initializations.append(self._configure_additional_transformers)
+
+    def check(self, _):
+        # type: (Any) -> None
+
+        self._init_base_tags()
+
+        if self.kubevirt_api_healthz_endpoint:
+            self._report_health_check(self.kubevirt_api_healthz_endpoint)
+        else:
+            self.log.warning(
+                "Skipping health check. Please provide a `kubevirt_api_healthz_endpoint` to ensure the health of the KubeVirt API."  # noqa: E501
+            )
+
+        self._validate_metrics_endpoint(self.kubevirt_api_metrics_endpoint)
```

Comment:
```suggestion
        self._validate_metrics_endpoint(self.kubevirt_api_metrics_endpoint)
```
This can happen during check initialization so do extra work every check run